Synapse3.SynapseModule/Events/PlayerEvents.cs

```diff
@@ -491,3 +493,14 @@ public StartWorkStationEvent(SynapsePlayer player, bool allow, SynapseWorkStatio
     public SynapseWorkStation WorkStation { get; }
 }
 
+public class FallingIntoAbyssEvent : IEvent
+{
+    public FallingIntoAbyssEvent(SynapsePlayer player, bool allow)
+    {
+        Player = player;
+        Allow = allow;
+    }
```

Comment:
Instead of inheriting from `IEvent` would it be better to inherit from `PlayerEvent` or `PlayerInteractEvent` so that a Plugin can hook multiple Player Events to the same method

```suggestion
public class FallingIntoAbyssEvent : PlayerInteractEvent
{
    public FallingIntoAbyssEvent(SynapsePlayer player, bool allow) : base(player, allow) { }
```
```suggestion
public class FallingIntoAbyssEvent : PlayerEvent
{
    public FallingIntoAbyssEvent(SynapsePlayer player, bool allow) : base(player)
    {
        Allow = allow;
    }
```